docs/populators.md

```diff
@@ -0,0 +1,171 @@
+
+## 2. Helpers
+
+In `stac-populator`, an extension is typically applied using an associated helper. A helper extends the [`Helper`](../STACpopulator/extensions/base.py#L64) abstract class and implements methods for retrieving or computing the data properties to be applied by the extension. For convenience an [`ExtensionHelper`]((../STACpopulator/extensions/base.py#L78)) class extends both `Helper` and the pydantic `BaseModel` classes to provide default constructor and validation mechanisms for defined attributes. 
+
+**A concrete helper** should extends `ExtensionHelper` and implement the following:
+
+1. Define property methods typically annotated with `@functools.cached_property`.
+2. Redefine as needed the `def apply(...)` method where data properties are applied using the corresponding STAC extension.
+3. Implement a `def from_data(...)` method to enable static initialization of the helper with extra kwargs.
+
+The example below shows an excerpt of the concrete `FileHelper` class. In addition to the required `data` field to compute the file `size` property, additional keyword arguments such as the http `session` can be passed in for initialization through `from_data(..., **kwargs)`. In particular, this static method facilitates instantiation of helpers at the next layer, i.e. in the `DataModel` associated with the populator implementation.
+
+```python
+import pystac
+import functools
+from requests import Session
+from pydantic import Field, Any
+from pystac.extensions.file import FileExtension
+
+from STACpopulator.extensions.base import ExtensionHelper
+
+
+class FileHelper(ExtensionHelper):
+    access_urls: dict[str, str]
+    ...
+    session: Optional[Session] = Field(default=None, exclude=True)
+
+    @classmethod
+    def from_data(
+        cls,
+        data: dict[str, Any],
+        **kwargs,
+    ) -> "FileHelper":
+        """Create a FileHelper instance from raw data."""
+        return cls(access_urls=data["data"]["access_urls"], **kwargs)
+
+    def apply(self, item: pystac.Item, add_if_missing: bool = True) -> T:
+        asset = item.assets[...]
+        file_ext = FileExtension.ext(asset, add_if_missing=add_if_missing)
+        file_ext.apply(
+            size=self.size,
+            ...
+        )
+
+    @functools.cached_property
+    def size(self) -> Optional[int]:
+        """Return file size in bytes, None if an error occurs."""
+        size: int = ... # Code logic to calculate the file size
+        return size
+```
+
+## 3. Data Model
+
+Inheriting from [`BaseSTAC`](../STACpopulator/extensions/base.py#L159), a `DataModel` is the main data management class associated with an implementation in `stac-populator`. It holds `helper` objects as attributes, which it uses to apply multiple extensions to a given entity (i.e., `item`, `asset`, `link`, or `collection`).
```

Comment:
cross-ref `helper` and `extensions` to previous sections

cross-ref the STAC objects with references (see one of my previous comment)